### ISSUE TITLE
Add dynamic master key

### ISSUE THREAD

**Issue by mtrezza** [parse-community/parse-dashboard#2580]
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Current Limitation

The master key is set on dashboard start and cannot be changed without a dashboard restart. This is disadvantageous when periodically rotating the keys because a key rotation also requires a dashboard restart.

### Feature / Enhancement Description

- Allow to set an async function for the master key instead of a string.
- Add the master key to the internal cache with a TTL, so that the function is not called every time the master key is validated

### Example Use Case

New Parse Dashboard options:
```
{
  masterKey: async () => await getKey(),
  masterKeyTtl: 3600
}
```

### Alternatives / Workarounds

Dashboard restart.

**Comment by dblythy** [comment]
Should this have a caching mechanism as well?

**Comment by mtrezza** [comment]
You are right, I removed that from the initial description, but re-added it now.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-alpha.29](https://github.com/parse-community/parse-dashboard/releases/tag/6.0.0-alpha.29)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.1.0](https://github.com/parse-community/parse-dashboard/releases/tag/6.1.0)
